react/finger-exercises/redux/02 - react-redux/src/app/screens/App/components/ShoppingCart/index.js

```diff
@@ -25,26 +26,30 @@ class ShoppingCart extends PureComponent {
   };
 
   render() {
-    const { data } = this.props;
+    const { bookSelected } = this.props;
     return (
       <Fragment>
         <Button className={styles.buttonCart} onClick={this.toggleContent}>
           <i className="fa fa-shopping-cart" />
         </Button>
         <div className={`${styles.container} ${this.state.open ? styles.open : ''}`}>
           <h1 className={styles.title}>Cart</h1>
-          <ul className={styles.content}>{data.map(this.renderItem)}</ul>
-          <h2 className={`${styles.title} ${styles.total}`}>Total: {data.reduce(this.total, 0)}</h2>
+          <ul className={styles.content}>{bookSelected.map(this.renderItem)}</ul>
+          <h2 className={`${styles.title} ${styles.total}`}>Total: {bookSelected.reduce(this.total, 0)}</h2>
         </div>
       </Fragment>
     );
   }
 }
 
 ShoppingCart.propTypes = {
-  data: arrayOf(bookSelectedPropType).isRequired,
+  bookSelected: arrayOf(bookSelectedPropType).isRequired,
   addItem: func.isRequired,
   removeItem: func.isRequired
 };
 
-export default ShoppingCart;
+const mapStateToProps = state => ({
+  bookSelected: state.bookSelected
+});
+
+export default connect(mapStateToProps)(ShoppingCart);
```

Comment:
you could also add mapDispatchToProps, with addItem and removeItem, instead of declaring them in the father component